modules/schedule_module/jsx/scheduleIndex.js

```diff
@@ -0,0 +1,584 @@
+/* eslint-disable */
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+import Modal from 'Modal';
+import swal from 'sweetalert2';
+import {Tabs, TabPane} from 'Tabs';
+import DataTable from 'jsx/DataTable';
+import Filter from 'jsx/Filter';
+/**
+ * Schedule Module
+ *
+ * Main module component rendering the data release module
+ *
+ * @author Shen Wang
+ */
+class ScheduleIndex extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      error: false,
+      isLoaded: false,
+      formData: {
+        AppointmentID: null,
+        StartsAt: null,
+        DCCID: null,
+        PSCID: null,
+        Session: null,
+        AppointmentDate: null,
+        AppointmentTime: '00:00',
+        AppointmentType: {},
+        SessionFieldOptions: {},
+      },
+      filter: {},
+      tabledatapast: {},
+      tabledatanext: {},
+      tabledatatoday: {},
+      showModal: false,
+      editModal: false,
+    };
+    this.updateFilter = this.updateFilter.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.mapColumn = this.mapColumn.bind(this);
+    this.renderScheduleForm = this.renderScheduleForm.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.fetchDataForm = this.fetchDataForm.bind(this);
+    this.edit = this.edit.bind(this);
+    this.renderScheduleFormButton = this.renderScheduleFormButton.bind(this);
+    this.deleteid = this.deleteid.bind(this);
+    this.tabByid = this.tabByid.bind(this);
+    this.deleteConfirm = this.deleteConfirm.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
```

Comment:
Hi @kongtiaowang, probably good to include the type of request "method" like GET or POST. Also  checking if the resp.ok is valid like here: https://github.com/aces/Loris/blob/1d41c87c18dc8947cfb9fbc15f54e237c3486303/modules/create_timepoint/jsx/createTimepointIndex.js#L125
I believe @xlecours or @zaliqarosli told me the catch doesn't always handle an unsuccessful response and why `resp.ok` should be used.